catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityFilterPlugin.java

```diff
@@ -104,17 +108,20 @@ public PolicyResponse processPostQuery(Result input, Map<String, Serializable> p
         Metacard metacard = input.getMetacard();
         HashMap<String, Set<String>> securityMap = new HashMap<>();
 
-        if ((metacard.getAttribute(Validation.VALIDATION_ERRORS) != null && metacard.getAttribute(
-                Validation.VALIDATION_ERRORS)
-                .getValues() != null) || (
-                metacard.getAttribute(Validation.VALIDATION_WARNINGS) != null &&
-                        metacard.getAttribute(Validation.VALIDATION_WARNINGS)
-                                .getValues() != null)) {
+        if (filterErrors && metacard.getAttribute(Validation.VALIDATION_ERRORS) != null &&
+                metacard.getAttribute(Validation.VALIDATION_ERRORS)
+                        .getValues() != null) {
+            for (Map.Entry<String, List<String>> attributeMapping : attributeMap.entrySet()) {
```

Comment:
You could probably even just pass the boolean into that `isAttributeSet` method and simplify more...
```java
        if (isAttributeSet(filterErrors, metacard, Validation.VALIDATION_ERRORS) || isAttributeSet(
                filterWarnings,
                metacard,
                Validation.VALIDATION_WARNINGS)) {
            for (Map.Entry<String, List<String>> attributeMapping : attributeMap.entrySet()) {
                securityMap.put(attributeMapping.getKey(),
                        new HashSet<>(attributeMapping.getValue()));
            }
        }
...
    private boolean isAttributeSet(boolean filterFlag, Metacard metacard, String attributeKey) {
        return filterFlag && metacard.getAttribute(attributeKey) != null && metacard.getAttribute(
                attributeKey)
                .getValues() != null;
    }
```